types/k6/crypto/index.d.ts

```diff
@@ -215,165 +215,4 @@ export abstract class Hasher {
-     * @returns SHA-384 digest.
-     * @example
-     * crypto.sha384('hello world!', 'hex')
-     */
-    function sha384<OE extends OutputEncoding>(input: string | ArrayBuffer, outputEncoding: OE): Output<OE>;
-
-    /**
-     * Hash with SHA-512.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/sha512/
-     * @param input - Data to hash.
-     * @param outputEncoding - Output encoding.
-     * @returns SHA-512 digest.
-     * @example
-     * crypto.sha512('hello world!', 'hex')
-     */
-    function sha512<OE extends OutputEncoding>(input: string | ArrayBuffer, outputEncoding: OE): Output<OE>;
-
-    /**
-     * Hash with SHA-512/224.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/sha512_224/
-     * @param input - Data to hash.
-     * @param outputEncoding - Output encoding.
-     * @returns SHA-512/224 digest.
-     * @example
-     * crypto.sha512_224('hello world!', 'hex')
-     */
-    function sha512_224<OE extends OutputEncoding>(input: string | ArrayBuffer, outputEncoding: OE): Output<OE>;
-
-    /**
-     * Hash with SHA-512/256.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/sha512_256/
-     * @param input - Data to hash.
-     * @param outputEncoding - Output encoding.
-     * @returns SHA-512/256 digest.
-     * @example
-     * crypto.sha512_256('hello world!', 'hex')
-     */
-    function sha512_256<OE extends OutputEncoding>(input: string | ArrayBuffer, outputEncoding: OE): Output<OE>;
-
-    /**
-     * Hash with RIPEMD-160.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/ripemd160/
-     * @param input - Data to hash.
-     * @param outputEncoding - Output encoding.
-     * @returns RIPEMD-160 digest.
-     * @example
-     * crypto.ripemd160('hello world!', 'hex')
-     */
-    function ripemd160<OE extends OutputEncoding>(input: string | ArrayBuffer, outputEncoding: OE): Output<OE>;
-
-    /**
-     * Create a hashing object.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/createhash/
-     * @param algorithm - Hash algorithm.
-     * @returns Hashing object.
-     * @example
-     * let hasher = crypto.createHash('sha256');
-     * hasher.update('hello ');
-     * hasher.update('world!');
-     * console.log(hasher.digest('hex'));
-     */
-    function createHash(algorithm: Algorithm): Hasher;
-
-    /**
-     * Create an HMAC hashing object.
-     * https://grafana.com/docs/k6/latest/javascript-api/k6-crypto/createhmac/
-     * @param algorithm - Hash algorithm.
-     * @param secret - Shared secret.
-     * @returns HMAC hashing object.
-     * @example
-     * let hasher = crypto.createHMAC('sha256', 'a secret');
-     * hasher.update('hello ');
-     * hasher.update('world!');
-     * console.log(hasher.digest('hex'));
-     */
-    function createHMAC(algorithm: Algorithm, secret: string | ArrayBuffer): Hasher;
-}
-
-export default crypto;
+export * as default from "k6/crypto";
```

Comment:
I don't fully get this? Where are those types defined now? 🤔

Reply:
Above in this same file :) 

Previously we defined them twice - once in the outside so we can support named imports `import {randomBytes} ..` but also defined a namespace that we then reexported as the default.

That not only requires that you copy stuff, but also namespaces make it so that they are kind of usable globally wihtout importing.  Neither one of which si a good idea.

This code just reexports the thigns defined as the default export as well. Which is basically what k6 does internally too.

I did try other ways to make this happen ... but none of them worked in all cases and this is actually pretty short and compact so :shrug:

Reply:
Ah, right, right! Apologies, I think got confused because the mix of how diffs are displayed + not being yet very used to this repository. Thanks for the explanation, all clear now!